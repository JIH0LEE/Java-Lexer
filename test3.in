int num=-3
char ch='a'
string myname="jiho    "
boolean flag=true
boolean flag2=false