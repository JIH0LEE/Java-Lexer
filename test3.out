<VTYPE,int>
<ID,num>
<ASSIGN>
<INTEGER,-3>
<VTYPE,char>
<ID,ch>
<ASSIGN>
<CHARACTER,'a'>
<VTYPE,string>
<ID,myname>
<ASSIGN>
<STRING,"jiho    ">
<VTYPE,boolean>
<ID,flag>
<ASSIGN>
<BOOLEAN,true>
<VTYPE,boolean>
<ID,flag2>
<ASSIGN>
<BOOLEAN,false>
