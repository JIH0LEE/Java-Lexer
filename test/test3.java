int summation(int n){
    int result=0;
    while(true){
        result=result+n;
    }
    if(true>=true){
        result=result+n;
    }
    else{
        result=result+n;
    }
    
    return result;
}