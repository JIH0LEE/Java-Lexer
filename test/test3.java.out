<1,VTYPE,int>
<1,ID,summation>
<1,LPAREN,(>
<1,VTYPE,int>
<1,ID,n>
<1,RPAREN,)>
<1,LBRACE,{>
<2,VTYPE,int>
<2,ID,result>
<2,ASSIGN,=>
<2,NUM,0>
<2,SEMI,;>
<3,WHILE,while>
<3,LPAREN,(>
<3,BOOLSTR,true>
<3,RPAREN,)>
<3,LBRACE,{>
<4,ID,result>
<4,ASSIGN,=>
<4,ID,result>
<4,OP,+>
<4,ID,n>
<4,SEMI,;>
<5,RBRACE,}>
<6,IF,if>
<6,LPAREN,(>
<6,BOOLSTR,true>
<6,COMP,>=>
<6,BOOLSTR,true>
<6,RPAREN,)>
<6,LBRACE,{>
<7,ID,result>
<7,ASSIGN,=>
<7,ID,result>
<7,OP,+>
<7,ID,n>
<7,SEMI,;>
<8,RBRACE,}>
<9,ELSE,else>
<9,LBRACE,{>
<10,ID,result>
<10,ASSIGN,=>
<10,ID,result>
<10,OP,+>
<10,ID,n>
<10,SEMI,;>
<11,RBRACE,}>
<13,RETURN,return>
<13,ID,result>
<13,SEMI,;>
<14,RBRACE,}>
